datasets/chicago_crime/_images/run_csv_transform_kub/csv_transform.py

```diff
@@ -0,0 +1,222 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import datetime
+import logging
+import math
+import os
+import pathlib
+import typing
+
+import pandas as pd
+import requests
+from google.cloud import storage
+
+
+def main(
+    source_url: str,
+    source_file: pathlib.Path,
+    target_file: pathlib.Path,
+    target_gcs_bucket: str,
+    target_gcs_path: str,
+) -> None:
+
+    logging.info(
+        "chicago crime process started at "
+        + str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"))
+    )
+
+    logging.info("creating 'files' folder")
+    pathlib.Path("./files").mkdir(parents=True, exist_ok=True)
+
+    logging.info(f"Downloading file {source_url}")
+    download_file(source_url, source_file)
+
+    logging.info(f"Opening file {source_file}")
+    df = pd.read_csv(source_file)
+
+    logging.info(f"Transforming {source_file} ...")
+
+    logging.info(f"Transform: Rename columns {source_file} ...")
+    rename_headers(df)
+
+    logging.info("Transform: Converting date format.. ")
+    convert_values(df)
+
+    logging.info("Transform: Removing null values.. ")
+    filter_null_rows(df)
+
+    logging.info("Transform: Reordering headers..")
+    df = df[
+        [
+            "unique_key",
+            "case_number",
+            "date",
+            "block",
+            "iucr",
+            "primary_type",
+            "description",
+            "location_description",
+            "arrest",
+            "domestic",
+            "beat",
+            "district",
+            "ward",
+            "community_area",
+            "fbi_code",
+            "x_coordinate",
+            "y_coordinate",
+            "year",
+            "updated_on",
+            "latitude",
+            "longitude",
+            "location",
+        ]
+    ]
+
+    logging.info("Transform: converting to integers..")
+
+    df["unique_key"] = df["unique_key"].apply(convert_to_integer_string)
+    df["beat"] = df["beat"].apply(convert_to_integer_string)
+    df["district"] = df["district"].apply(convert_to_integer_string)
+    df["ward"] = df["ward"].apply(convert_to_integer_string)
+    df["community_area"] = df["community_area"].apply(convert_to_integer_string)
+    df["year"] = df["year"].apply(convert_to_integer_string)
```

Comment:
[Optional] You can use a loop so it's more succinct.

Reply:
done